src/main/scala/mesosphere/marathon/storage/repository/GcActor.scala

```diff
@@ -141,7 +165,11 @@ private[storage] trait ScanBehavior[K, C, S] extends StrictLogging { this: FSM[S
           scan().pipeTo(self)
           goto(Scanning) using UpdatedEntities()
         } else {
-          goto(Idle) using IdleData
+          if (cleaningInteveral <= 0.millis) {
+            goto(ReadyForGc) using IdleData
+          } else {
+            goto(Resting) using IdleData
```

Comment:
nit: why is it still called `IdleData` when there is no `Idle` state?
nit2: State names are single words - wdyt about `Ready` instead of `ReadyForGc`?